### DISCUSSION TITLE
[slider] unexpected result on value initialization in javascript

### DISCUSSION THREAD

**Discussion by foutesting** [fomantic/Fomantic-UI#1705] [Modules (Modal, Dropdown, Popup, Calendar, Toast, ...)] [answered]
# Bug Report
Hello,
When I try to initialize the slider cursor with a number (eg 5000) in javascript with the Behaviors : set value(value), it doesn't work. 
Result: the slider goes to its maximum and there is a scale change.

## Steps to reproduce

1. HTML

<div class="mt-4">
<div class="ui red medium labeled text-gray-700 slider gaz" id="slider-consogaz"></div>
<input type="hidden" id="inp-consogaz" name="_consumptionkwhan" value="0">
</div>

2. JAVASCRIPT
$('.ui.slider.gaz')
  .slider(
    'set value', '5000',
    {
    min: 0,
    max: 30000,
    start: 250,
    step: 250,
    smooth: true,
            onChange: function(value) {
              $('#display-consogaz').html(value);
              document.getElementById("inp-consogaz").value = value;
       }
    })
;

## Expected result

the right value with the right scale when loading the page

## Testcase

https://jsfiddle.net/s76Leof4/

## Version

fomantic-ui@2.8.7
jquery@3.3.1

**Comment by lubber-de** [comment, marked as answer]
Before using behaviors like `set value` you have to initialize the slider first.
See https://jsfiddle.net/lubber/qkxwmgpr/3/

**Comment by foutesting** [comment]
thank you very much for your answer and for the correction with the right syntax !
